### ISSUE TITLE
TimeZoneInfo.HasIanaId returning false for an IANA timezone on Linux

### ISSUE THREAD

**Issue by roji** [dotnet/runtime#58326]
Running on Ubuntu 21.04, dotnet SDK 6.0.100-rc.1.21379.2, TimeZoneInfo.HasIanaId is unexpectedly returning false.

System timezone information:

```
> timedatectl
               Local time: Sun 2021-08-29 11:03:06 CEST
           Universal time: Sun 2021-08-29 09:03:06 UTC
                 RTC time: Sun 2021-08-29 09:03:06
                Time zone: Europe/Berlin (CEST, +0200)
System clock synchronized: yes
              NTP service: active
          RTC in local TZ: no
```

Based on the sample GetIanaTimeZoneId method in [this blog article](https://devblogs.microsoft.com/dotnet/date-time-and-time-zone-enhancements-in-net-6/):

```c#
Console.WriteLine(TimeZoneInfo.Local); // (UTC+01:00) Central European Time (Berlin)
Console.WriteLine(TimeZoneInfo.Local.Id); // Europe/Berlin
Console.WriteLine(TimeZoneInfo.Local.HasIanaId); // false

if (!TimeZoneInfo.TryConvertWindowsIdToIanaId(TimeZoneInfo.Local.Id, out _))
    Console.WriteLine($"Could not convert '{TimeZoneInfo.Local.Id}'"); // Could not convert 'Europe/Berlin'
```

**Comment by tarekgh** [comment]
Fixed through https://github.com/dotnet/runtime/pull/58414

@roji thanks for reporting this issue. I hope when you'll have a chance to try the fix. Let me know if there is anything I can help with.

**Comment by roji** [comment]
Thanks for the quick turnaround @tarekgh, happy this got fixed for 6.0! Will let you know if I run into anything else.
